openedx/features/enterprise_support/utils.py

```diff
@@ -301,26 +301,41 @@ def get_enterprise_learner_portals(request):
     Caches and returns results in/from the user's request session if provided.
     """
     # Prevent a circular import.
-    from openedx.features.enterprise_support.api import enterprise_enabled
+    from openedx.features.enterprise_support.api import enterprise_enabled, enterprise_customer_uuid_for_request
 
     user = request.user
     # Only cache this if a learner is authenticated (AnonymousUser exists and should not be tracked)
+
     if enterprise_enabled() and ENTERPRISE_HEADER_LINKS.is_enabled() and user and user.id:
         # If the key exists return that value
         if 'enterprise_learner_portals' in request.session:
             return json.loads(request.session['enterprise_learner_portals'])
-        # Ordering is important, this is consistent with how we decide on which
-        # enterprise_customer is the selected one for an enterprise_customer
-        enterprise_learner_portals = [{
-            'name': enterprise_customer_user.enterprise_customer.name,
-            'slug': enterprise_customer_user.enterprise_customer.slug,
-            'logo': enterprise_customer_user.enterprise_customer.branding_configuration.logo.url if
-            enterprise_customer_user.enterprise_customer.branding_configuration else None,
-        } for enterprise_customer_user in EnterpriseCustomerUser.objects.filter(
-            user_id=user.id, enterprise_customer__enable_learner_portal=True
-        ).prefetch_related(
-            'enterprise_customer', 'enterprise_customer__branding_configuration'
-        ).order_by('-enterprise_customer__active', '-modified')]
+
+        enterprise_customer_uuid = enterprise_customer_uuid_for_request(request)
+        kwargs = {
+            'user_id': user.id,
+            'enterprise_customer__enable_learner_portal': True,
+        }
+        if enterprise_customer_uuid:
+            kwargs['enterprise_customer__uuid'] = enterprise_customer_uuid
+
+        queryset = EnterpriseCustomerUser.objects.filter(**kwargs).prefetch_related(
+            'enterprise_customer',
+            'enterprise_customer__branding_configuration',
+        )
+        enterprise_learner_portals = []
```

Comment:
Do we need the list and for loop if we're only displaying 1 link?

Reply:
That's a good point.  We might not know the `enterprise_customer_uuid`, so it could actually return more than 1, I guess.  So I think what I need to do is:
* Add that ordering back in.
* Keep using the `filter()`, but tack on a `first()` so we really, really, will get at most 1 enterprise customer.